src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,360 @@
+    }
+
+    private List<String> getOnlinePlayerNames(@NotNull final MinecraftServer server)
+    {
+        final String[] onlinePlayerNames = server.getOnlinePlayerNames();
+        return Arrays.asList(onlinePlayerNames);
+    }
+
+    private List<String> getColonyIdStrings()
+    {
+        final List<Colony> colonyList = ColonyManager.getColonies();
+        final List<String> colonyIdList = new ArrayList<>(colonyList.size());
+        for (final Colony colony : colonyList)
+        {
+            colonyIdList.add(String.valueOf(colony.getID()));
+        }
+        return colonyIdList;
+    }
+
+    private List<String> getCitizenNames()
+    {
+        final List<Colony> colonyList = ColonyManager.getColonies();
+        final List<String> citizenNameList = new ArrayList<>();
+        for (final Colony colony : colonyList)
+        {
+            final List<CitizenData> citizenDataList = colony.getCitizenManager().getCitizens();
+            for (final CitizenData citizenData : citizenDataList)
+            {
+                citizenNameList.add(citizenData.getName());
+            }
+        }
+        return citizenNameList;
+    }
+
+    public List<String> getTabCompletions(@NotNull final MinecraftServer server,
+            @Nullable final BlockPos pos,
+            final String potentialArgumentValue)
+    {
+        switch (this)
+        {
+            case INTEGER:
+                return Collections.emptyList();
+            case BOOLEAN:
+                return Arrays.asList(new String[] {"true", "false"});
+            case COORDINATE_X:
+            case COORDINATE_Y:
+            case COORDINATE_Z:
+                if (potentialArgumentValue.isEmpty())
+                {
+                    switch (this)
+                    {
+                        case COORDINATE_X:
+                            return Collections.singletonList(String.valueOf(pos.getX()));
+                        case COORDINATE_Y:
+                            return Collections.singletonList(String.valueOf(pos.getY()));
+                        case COORDINATE_Z:
+                            return Collections.singletonList(String.valueOf(pos.getZ()));
+                        default:
+                            // We will never reach here.
+                            break;
+                    }
+                    return Collections.emptyList();
+                }
+                try
+                {
+                    Integer.parseInt(potentialArgumentValue);
+                    return Collections.emptyList();
+                }
+                catch (final NumberFormatException e)
+                {
+                    return Collections.emptyList();
+                }
+            case PLAYER:
+                final List<String> playerNameStrings = getOnlinePlayerNames(server);
+                return playerNameStrings.stream().filter(k -> k.startsWith(potentialArgumentValue)).collect(Collectors.toList());
+            case COLONY:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 14 to at most 10, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1151)